api/producer/set-minimum-version.go

```diff
@@ -0,0 +1,36 @@
+/*
+   Copyright © 2025 The CDI Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package producer
+
+import (
+	"fmt"
+
+	cdi "tags.cncf.io/container-device-interface/specs-go"
+)
+
+type setMinimumRequiredVersion struct{}
+
+// Transform detects the minimum required version required for the specified
+// spec and sets the version field accordingly.
+func (d setMinimumRequiredVersion) Transform(spec *cdi.Spec) error {
```

Comment:
Is this an external API?

Reply:
It's not required to be public here. In the NVIDIA Container Toolkit implementation, we have the following interface:
```
type Transformer interface {
    Transform(*cdi.Spec) error
}
```
which allows for composition. 

I reused some of the implementation details, and didn't change the function visibility. 

Note that since the type itself is private, defining the function as public should not affect the API at all.

Reply:
Yes, but it creates confusion. At least I was confused. Can we make it non-public?

Reply:
I will update it.